src/workerinfo.js

```diff
@@ -34,57 +35,65 @@ class WorkerInfo {
     }
   }
 
-  async seen(provisionerId, workerType) {
+  async seen(provisionerId, workerType, workerGroup, workerId) {
     const expired = entity => Date.now() - new Date(entity.expires) > 24 * 60 * 60 * 1000;
     const expires = taskcluster.fromNow('5 days');  // temporary hard coded expiration
 
+    const updateExpiration = (entry) => {
+      entry.modify(entity => {
+        if (expired(entity)) {
+          entity.expires = expires;
+        }
+      });
+    };
+
+    const createEntry = async (entity, entry) => {
```

Comment:
@jonasfj From my understanding, the `await` operator returns the resolved value of a promise so chaining the catch method will probably not work. I think we'll have to use try and catch statements:
```js
const createEntry = async (entity, entry) => {
  try {
    await entity.create(entry);
  } catch (err) {
    ignoreEntityAlreadyExists(err);
  }
};
```